### PR TITLE
Add docker publish

### PR DESCRIPTION
**Pull request by vemonet**
Hi @huberrob I made some improvement to automatically publish a Docker image

* fixed an error when running the FUJI API in Docker by upgrading `werkzeug~=1.0.0` to `werkzeug~=2.0`. You might want to check locally or your tests to make sure it does not break anything, the API works fine through docker one my side.
*  Added a GitHub Action workflow to automatically build and publish a new image with the `latest` tag at each push to the `master` branch. It will also publish an image with a tag corresponding to the release version, when you publish a new release on GitHub (with id starting with v, e.g. `v0.1.0`)

You can pull and try the image published by the workflow under my fork:

```bash
docker run -it -p 1071:1071 ghcr.io/vemonet/fuji
```

You can checkout the workflow run here: https://github.com/vemonet/fuji/actions

After merging it will publish it as `ghcr.io/pangaea-data-publisher/fuji`, you can change the image name if you want in the workflow file in `.github/workflows/publish-docker.yml`. I already made sme changes to the README to include example with the image to be published under your organiztion

This feature has been mentioned in the issue https://github.com/pangaea-data-publisher/fuji/issues/151

Advantage of using GitHub Actions and Container Registry over DockerHub and quay.io:
* (Still) completely free for open source project (DockerHub introduced pull rates limits that can be a pain, especially when deploying on Kubernetes clusters)
* Everything is contained in your GitHub repository, no need for extra services/accounts, and authentication to publish the docker image is done automatically via the workflow GITHUB_TOKEN

### CODE REVIEW

**Comment by huberrob** [comment]
Great... @wilkos-dans can you please take a look at this as you provided the original docker code

File: README.md [hunk 1]
```diff
@@ -67,6 +70,18 @@ Your Swagger definition lives here:
 http://localhost:1071/uji/api/v1/swagger.json
```
Comment (wilkos-dans) [inline README.md:70]: Typo in URL? Should be:
http://localhost:1071/fuji/api/v1/ui/

**Comment by wilkos-dans** [comment]
> Great... @wilkos-dans can you please take a look at this as you provided the original docker code

Really nice feature! No comments from my side.

File: README.md (hunk 1 above)
Comment (huberrob) [reply README.md:70]: @vemonet Could you please check if this is a typo? See @wilkos-dans comment above

Comment (huberrob) [reply README.md:70]: @vemonet please forget my last comment

**Pull request merged**